### PR TITLE
Added type for @webpack-blocks/core

### PR DESCRIPTION
**Pull request by maxbogus**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/webpack-blocks__core/webpack-blocks__core-tests.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { createConfig, group, env, match, when, Block, Context, MatchOptions, Util } from '@webpack-blocks/core';
```
Comment (armanio123) [inline types/webpack-blocks__core/webpack-blocks__core-tests.ts]: Is the import correct? The documentation only uses `webpack-blocks` and not `@webpack-blocks/core`.

Comment (maxbogus) [reply types/webpack-blocks__core/webpack-blocks__core-tests.ts]: If you check [doc](https://www.npmjs.com/package/webpack-blocks) you will see following line: "This is the webpack-blocks convenience package. It wraps all the most commonly used blocks, so you can install just this single package and have webpack-blocks and your favorite blocks set up."

My plan:
1. Cover separate blocks (@webpack-blocks/core and so on)
2. Create main package with links to separate packages.

If we cover all functions in one package (webpack-blocks), we will leave without support subpackages (@webpack-blocks/core, @webpack-blocks/webpack and so on).

Comment (maxbogus) [reply types/webpack-blocks__core/webpack-blocks__core-tests.ts]: Technically, I can do this in one pull-request:
1. create several folders for (i.e. webpack-blocks__core and so on).
2. create one folder for webpack-blocks and import separate blocks.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
> Technically, I can do this in one pull-request:
>
> 1. create several folders for (i.e. webpack-blocks__core and so on).
> 2. create one folder for webpack-blocks and import separate blocks.

@maxbogus can you go ahead and do this? I want to make sure it doesn’t slip through the cracks. Thanks!

**Comment by maxbogus** [comment]
> > Technically, I can do this in one pull-request:
> > 
> > 1. create several folders for (i.e. webpack-blocks__core and so on).
> > 2. create one folder for webpack-blocks and import separate blocks.
> 
> @maxbogus can you go ahead and do this? I want to make sure it doesn’t slip through the cracks. Thanks!

I will! Today I'm planning to push everything to the origin.

**Comment by andrewbranch** [comment]
Oh, if it wasn’t clear, I just meant let’s do it as part of this PR—I wasn’t trying to say “hurry up and do it ASAP,” was just trying to make sure you weren’t waiting on this PR to get merged first 😅

**Review by andrewbranch — CHANGES_REQUESTED** [review]
`webpack-blocks`, `@webpack-blocks/core`, `@webpack-blocks/assets`, and `@webpack-blocks/webpack` are currently represented as ES2015 modules with named exports, when actually they’re CommonJS modules that export an object. The proper way to declare this is by putting all the exported things in a namespace, and then writing an `export =` for that namespace. Example:

Change this:
```ts
export function createConfig(configSetters: Block[]): Block;
export function group(configSetters: Block[]): Block;
// ...
```

to this:

```ts
declare namespace Core {
  function createConfig(configSetters: Block[]): Block;
  function group(configSetters: Block[]): Block;
  // ...
}

export = Core;
```

The end result is you shouldn’t have any ES2015-style imports or exports, because the actual JS is written with just `module.exports =`. Make sense?

**Review by andrewbranch — APPROVED** [review]
Update: sorry, my previous review was just wrong—those are equivalent and the linter will make you do the way you have it now. Sorry for the confusion!

**Pull request merged**